File: Codeforces root folder 3/Kousuke's Assignment.cpp
// TLE
#include<bits/stdc++.h>
using namespace std;

void solve() {
    int n; cin >> n;
    int nums[n];
    for (int i = 0; i < n; i++) cin >> nums[i];

    unordered_map<long long, int> mp;
    mp[0] = 1;
    deque<long long> prefix;
    prefix.push_back(0);
    int res = 0;
    long long total = 0;

    for (int num : nums) 
    {
        total += (long long)num;
        if (mp.find(total) != mp.end()) 
        {
            res++;
            while (prefix.size()) {
                long long cur = prefix.front();
                prefix.pop_front();
                mp[cur]--;
                if (mp[cur] == 0) mp.erase(cur);
            }
        }
        mp[total]++;
        prefix.push_back(total);
    }
    cout << res << endl;
}


int main() {
    cin.tie(nullptr);
    int t; cin >> t;
    for (int i = 0; i < t; i++) {
        solve();
    }
}

File: Codeforces root folder 3/Kousuke's Assignment.py
# hashing solution: TLE
from collections import defaultdict, deque, Counter
import heapq, math


def solve():
    n = int(input())
    nums = [int(e) for e in input().split()]
    total = 0
    mp = defaultdict(int)
    mp[0] = 1
    prefix = deque([0])
    res = 0
    for num in nums:
        total += num

        if total in mp:
            # print(total)
            res += 1
            while prefix:
                cur = prefix.popleft()
                mp[cur] -= 1
                if mp[cur] == 0:
                    mp.pop(cur)
        mp[total] += 1
        prefix.append(total)

    print(res)

t = int(input())
for i in range(t):
    solve()


